datadog/api/http_client.py

```diff
@@ -85,4 +88,78 @@ class URLFetchClient(HTTPClient):
     """
     HTTP client based on Google App Engine `urlfetch` module.
     """
-    pass
+    @classmethod
+    def request(cls, method, url, headers, params, data, timeout, proxies, verify, max_retries):
+        """
+        Wrapper around `urlfetch.fetch` method.
+
+        TO IMPLEMENT:
+        * `max_retries`
+        """
+        # No local certificate file can be used on Google App Engine
+        validate_certificate = True if verify else False
+
+        # Encode parameters in the url
+        url_with_params = "{url}?{params}".format(
+            url=url,
+            params=urllib.urlencode(params)
+        )
+
+        try:
+            result = urlfetch.fetch(
+                url=url_with_params,
+                method=method,
+                headers=headers,
+                validate_certificate=validate_certificate,
+                deadline=timeout,
+                payload=data
+            )
+
+            cls.raise_on_status(result)
+
+        except urlfetch.DownloadError as e:
+            raise ClientError(method, url, e)
+        except urlfetch_errors.DeadlineExceededError:
+            raise HttpTimeout(method, url, timeout)
+
+        return result
+
+    @classmethod
+    def raise_on_status(cls, result):
+        """
+        Raise on HTTP status code errors.
+        """
+        status_code = result.status_code
+
+        if (status_code / 100) != 2:
+            if status_code in (400, 403, 404, 409):
+                pass
+            else:
+                raise HTTPError(status_code)
+
+
+HTTP_CLIENTS = [RequestClient, URLFetchClient]
+
+
+def get_http_client():
+    """
+    Return the appropriate HTTP client based the defined priority and user environment.
+    """
+    global requests
+    global urlfetch
+    global urlfetch_errors
+
+    try:
+        import requests
```

Comment:
I'm not a huge fan of this solution (use of `global` + `import` inside a method).
Why not `import` with `try...except` at the top of the file, defaulting to None, and then here just check for `None` or not ?